runtime/commonMain/src/kotlinx/serialization/protobuf/ProtoBuf.kt

```diff
@@ -40,6 +40,9 @@ class ProtoBuf(context: SerialModule = EmptyModule) : AbstractSerialFormat(conte
             else -> throw SerializationException("Primitives are not supported at top-level")
         }
 
+        override fun shouldWriteNullElement(desc: SerialDescriptor, tag: ProtoDesc, index: Int): Boolean =
+            desc.kind !is StructureKind.CLASS
```

Comment:
This check is still not under the protobuf-specific configuration flag, so it would change already existing semantic. Also, I can imagine that it can be more flexible, like `configuration.nullSupportMode = DISABLED` will have old semantic, `configuration.nullSupportMode = DROP_NULLS` will drop `null` values to be compatible with another clients, and `configuration.nullSupportMode = DROP_DEFAULTS` will drop default values as in Json.

Reply:
Also, it makes sense to move `throw SerializationException("null is not supported")` somewhere here to insert more meaningful error message (with descriptor kind, current tag, etc) and how the user can change configuration (again, see `JsonConfiguration.allowStructuredMapKeys` implementation and how this flag is used)